### ISSUE TITLE
Uncaught TypeError: path must be a string

### ISSUE THREAD

**Issue by Krishath** [AtomLinter/linter-clang#46]
I just opened a new Atom window and the exception was thrown. How to reproduce it is simple as:

1. Open a new file
2. Set C as file specific programming language

**Atom Version**: 0.176.0
**System**: linux 3.16.0-29-generic
**Thrown From**: [linter-clang](https://github.com/AtomLinter/linter-clang) package, v2.22.0

### Stack Trace

Uncaught TypeError: path must be a string

```
At fs.js:691

TypeError: path must be a string
  at TypeError (native)
  at Object.fs.readdirSync (fs.js:691:18)
  at Object.fs.readdirSync (/usr/share/atom/resources/atom/common/lib/asar.js:398:26)
  at searchDirectory (/home/giulio/.atom/packages/linter-clang/lib/linter-clang.coffee:91:17)
  at LinterClang.lintFile (/home/giulio/.atom/packages/linter-clang/lib/linter-clang.coffee:120:5)
  at /home/giulio/.atom/packages/linter/lib/linter-view.coffee:139:18
  at Array.forEach (native)
  at /home/giulio/.atom/packages/linter/lib/linter-view.coffee:138:18
  at Object.oncomplete (fs.js:93:15)

```

### Commands

```
 13x -0:08.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:02.2 grammar-selector:show (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "themes": [
      "atom-dark-ui",
      "monokai"
    ]
  }
}
```

### Installed Packages

```coffee
# User
autocomplete-plus, v2.0.4
linter, v0.10.1
linter-clang, v2.22.0
linter-pep8, v0.1.0
linter-shellcheck, v0.0.6
minimap, v3.5.6
monokai, v0.12.0

# Dev
No dev packages
```

/cc @atom/core

**Comment by m-bra** [comment]
Okay okay okay, the third issue for the same bug now, it's okay, already fixed with #45 , has to be pulled yet! @Snake231088

**Comment by m-bra** [comment]
If you want to provide more information, reproduce while 1) in dev mode 2) debug option in linter-clang settings is checked. Then give the console output (ctrl/cmd - shift - I, select console tab)
